Code maintenance/61390 enable disabled specs in spec/features/activities/work_package/activities_spec.rb

akabiru opened the pull request:
https://community.openproject.org/wp/61390

The general gist is tweaking wait times- and adding retries to address flakiness

akabiru committed:
[#61390] Disabled specs in ./spec/features/activities/work_package/activities_spec.rb

https://community.openproject.org/work_packages/61390

akabiru committed:
Attempt to blindly re-enable commented out specs

This reverts commit 0b15a6a4fa25a8d3fb3a6d34caaf1921586b473d.

oliverguenther commented on spec/features/activities/work_package/activities_spec.rb:
```diff
@@ -494,94 +494,89 @@
+        activity_tab.expect_journal_notes(text: "Third comment by admin")
       end
+
+      def wait_for_filter = sleep(1)
```
this will not be a very stable spec, but I'm willing to merge it and see how flickering it ends up.

akabiru replied:
yeah, lm see if I can improve this to assert against the DOM

akabiru committed:
Remove static wait time on activities filter journal

Reinstate DOM enabled wait time introduced in https://github.com/opf/openproject/pull/17055

akabiru committed:
Switch to rspec wait from capybara max wait time

akabiru committed:
Rollout rspec wait to rest of activities

akabiru committed:
Enable disabled spec

akabiru committed:
Increase polling interval for activity tab assertions

akabiru committed:
Reinstate capybara max wait time for journal notes negated assertion

akabiru committed:
switch `wp_page.wait_for_activity_tab` to rspec wait

akabiru committed:
resolve Style/FrozenStringLiteralComment

akabiru committed:
Allow delay in filter journals

Allow additional waiting to give the menu time to close and page to load

akabiru committed:
Define constant 1s delay in journals filter to reduce flakiness
Give the page time to load.

akabiru commented on spec/features/activities/work_package_activity_spec.rb:
```diff
@@ -52,6 +52,11 @@
     before do
+      # set WORK_PACKAGES_ACTIVITIES_TAB_POLLING_INTERVAL_IN_MS from 10s to 1s
+      # to speed up the polling interval for test duration
+      # TODO: Redefine interval as a setting?
```
https://community.openproject.org/wp/61596

akabiru replied:
```suggestion
      # TODO: Redefine interval as a setting? [OP#61596]
```

akabiru committed:
Wait for action menu on click

akabiru committed:
Revert "Wait for action menu on click"

This reverts commit 3f2196814bbda0c8cb85e3d1ca860a1b4d9d9e6e.

akabiru committed:
reinstate static wait to allow for page load

akabiru committed:
Revert "Rollout rspec wait to rest of activities"

This reverts commit 992caa19ba4d34fc99e51b17a688387ff518d3e5.

akabiru committed:
retry journal finds to avoid flakiness

brunopagno commented on spec/features/activities/work_package_activity_spec.rb:
```diff
@@ -60,6 +65,10 @@
     end
 
+    after do
+      ENV.delete("WORK_PACKAGES_ACTIVITIES_TAB_POLLING_INTERVAL_IN_MS")
```
Shouldn't we do something like:

```ruby
temp_local_thingy = ENV["THINGY"]
ENV["THINGY"] = "1000"
...
ENV["THINGY"] = temp_local_thingy
```

I'm assuming the `delete`  would affect cases when someone has configured the environment variable manually.

Or it could accidentally leak behaviour to other tests that also rely on this ENV.

akabiru replied:
Yeah, it's odd looking but it should be fine as it has a configured fallback in https://github.com/opf/openproject/blob/dev/app/components/work_packages/activities_tab/index_component.rb#L74 I just followed the current convention, but in the end I'd like to see this as a configurable setting. https://community.openproject.org/wp/61596

brunopagno reviewed (approved):
Looks good. :tada: 

I am just unsure about the deleting the env variable step. Maybe I'm overthinking. I'll let you be the judge.
